### PR TITLE
JAMES-2340 Configure SpamAssassin mailet pipeline

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/IsMarkedAsSpam.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public void init() throws MessagingException {
+    }
+
+    @Override
+    public Collection<MailAddress> match(Mail mail) throws MessagingException {
+        Serializable attribute = mail.getAttribute(SpamAssassinInvoker.STATUS_MAIL_ATTRIBUTE_NAME);
+        if (isMarkedAsSpam(attribute)) {
+            return mail.getRecipients();
+        }
+        return new ArrayList<>();
```
Comment (mbaechler) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/IsMarkedAsSpam.java]: why an ArrayList ?

**Comment by aduprat** [comment]
> I don't understand the second commit. Why cannot you write an integration test doing the same thing (I mean launching a spamassassin in docker and configuring the mailet pipeline)

It's acting as both documentation & tool.
Having real integration test is really great, but you are not showing to customer how to configure James & define your mailet pipeline.

**Comment by rouazana** [comment]
> It's acting as both documentation & tool.
Having real integration test is really great, but you are not showing to customer how to configure James & define your mailet pipeline.

For me it's the exact goal of a good integration test. How do you check that your tool is still working in 6 months? 2 years?

**Comment by aduprat** [comment]
> For me it's the exact goal of a good integration test. How do you check that your tool is still working in 6 months? 2 years?
I have to check that it is working now.
It's not an integration test, but better than nothing.

**Comment by mbaechler** [comment]
> It's not an integration test, but better than nothing.

Or not, because eventually, the intent of this will be lost and we will have to find if we can drop it or not.
Adding things as a cost _in the future_

If it's for today, just run it and don't commit. Or write doc.

**Comment by aduprat** [comment]
Ok, I reverted the manual testing tool + added some documentation in the Matcher itself.

**Pull request merged**